File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        n = len(s)
        g.sort(reverse=True) # Sort g in descending order
        s.sort(reverse=True) # Sort s in descending order
        curr, total = 0, 0
        for i in range(len(g)):
            if curr < n and g[i] <= s[curr]:
                curr += 1
                total += 1
        return total

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int n = s.size();
        sort(g.rbegin(), g.rend());
        sort(s.rbegin(), s.rend());
        int curr=0, total =0;
        for(int i =0; i<g.size(); i++){
            if(curr<n and g[i]<= s[curr]){
                curr++;
                total++;
            }
        }
        return total;
    }
};